src/hint_processor/builtin_hint_processor/uint384.rs

```diff
@@ -289,8 +181,57 @@ pub fn uint384_signed_nn(
     insert_value_into_ap(vm, res)
 }
 
+/* Implements Hint:
+%{
+    def split(num: int, num_bits_shift: int, length: int):
+        a = []
+        for _ in range(length):
+        a.append( num & ((1 << num_bits_shift) - 1) )
+        num = num >> num_bits_shift
+        return tuple(a)
+
+    def pack(z, num_bits_shift: int) -> int:
+        limbs = (z.d0, z.d1, z.d2)
+        return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+
+    a = pack(ids.a, num_bits_shift = 128)
+    b = pack(ids.b, num_bits_shift = 128)
+    p = pack(ids.p, num_bits_shift = 128)
+
+    res = (a - b) % p
+
+
+    res_split = split(res, num_bits_shift=128, length=3)
+
+    ids.res.d0 = res_split[0]
+    ids.res.d1 = res_split[1]
+    ids.res.d2 = res_split[2]
+%}
+*/
+pub fn sub_reduced_a_and_reduced_b(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let a = Uint384::from_var_name("a", vm, ids_data, ap_tracking)?.pack();
+    let b = Uint384::from_var_name("b", vm, ids_data, ap_tracking)?.pack();
+    let p = Uint384::from_var_name("p", vm, ids_data, ap_tracking)?.pack();
+    let res = if a > b {
+        (a - b).mod_floor(&p)
+    } else if a < p {
+        &a + &p - &b
+    } else {
+        &p + &a.mod_floor(&p) - b.mod_floor(&p)
```

Comment:
This could also be expressed as:
```rust
&p - (b - &a).mod_floor(&p)
```
What do you think?

Reply:
sounds good